website/pages/en/cookbook/subgraph-composition.mdx

```diff
@@ -0,0 +1,125 @@
+---
+title: Subgraph Composition with Sushiswap v3 and Base
+---
+
+Use subgraph composition to create a base subgraph containing core data, facilitating the building and scaling of additional subgraphs on top.
+
+## Overview
+
+This guide uses the Sushiswap v3 subgraph on Base chain.
+
+### Benefits of Composition
+
+Subgraph composition is a powerful feature for scaling, enabling the reuse of existing data. It allows you to:
+
+- Speed up your subgraph's syncing speed
+- Handle errors and optimize the resync
+- Enhance your subgraph development to effortlessly scale
+
+> Note: You can use existing subgraphs, but they must be redeployed. This does not require you to write new code. However, you might want to separate your subgraph, which could require additional adjustments to development.
+
+## Setup
+
+### Tools Needed
+
+The setup involves two subgraphs:
+
+1. **Source Subgraph**: Tracks event data as entities.
+2. **Dependent Subgraph**: Uses the source subgraph as a data source.
+
+You can find these in the `source` and `dependent` directories.
+
+- The **source subgraph** is a basic event-tracking subgraph that records events emitted by relevant contracts.
+- The **dependent subgraph** references the source subgraph as a data source, using the entities from the source as triggers.
+
+While the source subgraph is a standard subgraph, the dependent subgraph utilizes the subgraph composition feature.
+
+### Source Subgraph
+
+The source subgraph tracks events from the Sushiswap v3 subgraph on the Base chain. This subgraph's configuration file is in the `source/subgraph.yaml`.
+
+### Dependent Subgraph
+
+The dependent subgraph is in the `dependent/subgraph.yaml`, which specifies the source subgraph as a data source. This subgraph uses entities from the source to trigger specific actions based on changes to those entities.
+
+## Get Started
+
+The following is a how-to-guide on using one subgraph as a data source for another.
+
+### Step 1. Set Up Your Source Subgraph
+
+To set the source subgraph as a data source in the dependent subgraph, include the following in `subgraph.yaml`:
+
+```yaml
+specVersion: 1.3.0
+schema:
+  file: ./schema.graphql
+dataSources:
+  - kind: subgraph
+    name: Factory
+    network: scroll
```

Comment:
The title says base and the network here is scroll. This is MY mistake the example subgraph had scroll in it, i just updated it on the example subgraph, would be great to update it accordingly here too.